tlatools/org.lamport.tlatools/src/tlc2/tool/EvalControl.java

```diff
@@ -81,4 +84,46 @@ public static boolean isInit(int control) {
 	public static boolean isConst(int control) {
 		return isSet(control, Const);
 	}
+
+    /**
+     * Determine whether two {@link EvalControl} settings are semantically equivalent.  Formally,
+     * they are semantically equivalent if
+     * <pre>
+     *     \A expr, context, behavior:
+     *         eval(expr, context, behavior, control1) =
+     *         eval(expr, context, behavior, control2)
+     * </pre>
+     *
+     * <p>Many control settings like {@link #Clear} and {@link #Enabled} are not equivalent; they
+     * will often result in different computed values.
+     *
+     * <p>This function always returns {@link PartialBoolean#YES} when
+     * <code>control1 == control2</code>, and it can identify a small set of other cases where
+     * the inputs are semantically equivalent.
+     *
+     * @param control1 the first control value
+     * @param control2 the second control value
+     * @return whether the values are semantically equivalent
+     */
+    public static PartialBoolean semanticallyEquivalent(int control1, int control2) {
+        // *** CAUTION ***
+        // The implementation of this function is quite subtle.  First we'll define `flagsThatCanBeSafelyIgnored`
+        // as a special whitelist of flags we know won't affect the evaluation outcome.  Each whitelisted flag
+        // has to be carefully justified:
+        //   - KeepLazy: this is a performance hint that affects handling of function definitions.  Although it
+        //     can change the exact structure of the IValue returned by eval(...), it does not affect the semantic
+        //     meaning of that value.
+        //   - Init: this is a flag to indicate that we are evaluating part of a specification's initial condition.
+        //     It is used for debugging.
```

Comment:
What do you mean by "It is used for debugging."?  This statement would be false if it refers to the TLA+ debugger.

Reply:
You are right, there are a few things wrong with this comment. "is used" is vague, and it's not obvious what I was getting at here.

I think when I wrote that I had just looked at every place in the code where control flow depends on `EvalControl.isInit` and `EvalControl.isConst`. There are (I think) exactly two:
 - `TLCGetSet`, where `isInit` affects how the level of the state is computed
 - `DebugTool`, where `isInit` helps set the correct value for the `mode` field

The second one is the one I intended to point at with this comment.

`Init` and `Const` are "used" lots of places (in the sense that they are set during evaluation). However, as far as I could tell, they are only "used" in a very small number of places (in the sense that their value is inspected).

Anyway, I'm open to suggestions for how this can be better.